patroni/postgresql/config.py

```diff
@@ -186,30 +317,82 @@ def replace_pg_ident(self):
                     f.write('{0}\n'.format(line))
             return True
 
-    def primary_conninfo(self, member):
+    def primary_conninfo_params(self, member):
         name = self._postgresql.name
         if not (member and member.conn_url) or member.name == name:
             return None
-        r = member.conn_kwargs(self.replication)
-        r.update(application_name=name, sslmode='prefer', sslcompression='1', krbsrvname=self._krbsrvname)
-        keywords = 'user password host port sslmode sslcompression application_name krbsrvname'.split()
+        ret = member.conn_kwargs(self.replication)
+        ret.update(application_name=name, sslmode='prefer')
+        if self._krbsrvname:
+            ret['krbsrvname'] = self._krbsrvname
+        if 'database' in ret:
+            del ret['database']
+        return ret
+
+    def primary_conninfo(self, member):
+        r = self.primary_conninfo_params(member)
+        if not r:
+            return None
+        keywords = 'user password host port sslmode application_name krbsrvname'.split()
         return ' '.join('{0}={{{0}}}'.format(kw) for kw in keywords if r.get(kw)).format(**r)
 
     def recovery_conf_exists(self):
         return os.path.exists(self._recovery_conf)
 
-    def check_recovery_conf(self, member):
+    def _read_primary_conninfo(self):
         # TODO: recovery.conf could be stale, would be nice to detect that.
```

Comment:
The PR resolves this TODO, does't it ?

Reply:
Nope, one can overwrite the recovery.conf and don't restart postgres. There is no way to check the actual value of primary_conninfo before pg12.